data_model/src/lib.rs

```diff
@@ -1800,19 +1803,11 @@ pub mod peer {
     }
 
     /// Peer's identification.
+    ///
+    /// Equality is tested by `public_key` field only.
+    /// Each peer should have a unique public key.
     #[derive(
-        Debug,
-        Clone,
-        Eq,
-        PartialEq,
-        PartialOrd,
-        Ord,
-        Hash,
-        Decode,
-        Encode,
-        Deserialize,
-        Serialize,
-        IntoSchema,
+        Debug, Clone, PartialOrd, Ord, Eq, Decode, Encode, Deserialize, Serialize, IntoSchema,
```

Comment:
what about ordering?

Reply:
PartialOrd requires PartialEq and we already have a custom impl of it, so I think it should be ok.

Reply:
Then it might be a good idea to make `public_key` be the first field of the struct. Just so that ordering is first done by it